importlib_metadata/_meta.py

```diff
@@ -35,14 +36,18 @@ class SimplePath(Protocol):
     A minimal subset of pathlib.Path required by PathDistribution.
     """
 
-    def joinpath(self) -> 'SimplePath':
+    def joinpath(self, *other: Union[str, 'PathLike[str]']) -> 'SimplePath':
         ...  # pragma: no cover
 
-    def __truediv__(self) -> 'SimplePath':
+    def __truediv__(self, other: Union[str, 'PathLike[str]']) -> 'SimplePath':
```

Comment:
`Path` uses `key` as the parameter name, `other` produces an error in Pyright:

```suggestion
    def __truediv__(self, key: Union[str, 'PathLike[str]']) -> 'SimplePath':
```

Reply:
(Arguably a bug in Pyright.)